moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.hpp

```diff
@@ -104,7 +104,7 @@ class TrajectoryVisualization : public QObject
   virtual void update(float wall_dt, float ros_dt);
   virtual void reset();
 
-  void onInitialize(Ogre::SceneNode* scene_node, rviz_common::DisplayContext* context, const rclcpp::Node& ros_node);
+  void onInitialize(Ogre::SceneNode* scene_node, rviz_common::DisplayContext* context, rclcpp::Node::SharedPtr ros_node);
```

Comment:
Why remove the const and the reference (&)?

Reply:
removing & because using SharedPtr is a common practice in ROS2. regarding the **const**, I find strange to use it with a node pointer, but will research this.

Note that the original moveit1code is `const ros::NodeHandle& update_nh`

Reply:
We usually leave it as a const, if it's already a const

Reply:
Added const to the SharedPtr
0fc4f6629d8e7ffadaa13200233f1c483bb1f241

Reply:
We are going with `ConstSharedPtr`, info here: https://design.ros2.org/articles/generated_interfaces_cpp.html

Resolving conversation.